src/Bridge/Doctrine/Orm/ItemDataProvider.php

```diff
@@ -94,13 +102,25 @@ public function getItem(string $resourceClass, $id, string $operationName = null
 
         $repository = $manager->getRepository($resourceClass);
         $queryBuilder = $repository->createQueryBuilder('o');
+        $i = 0;
+        $identifiersObj = new \ArrayObject($identifiers);
+        $identifiersObjIterator = $identifiersObj->getIterator();
+
+        while ($identifiersObjIterator->valid()) {
+            $placeholder = 'id_'.$identifiersObjIterator->key();
+            $whereExpr[$i] = $queryBuilder->expr()->eq('o.'.$identifiersObjIterator->key(), ':'.$placeholder);
+
+            if ($i <= 0) {
+                $queryBuilder
+                    ->where($whereExpr[$i]);
+            } else {
+                $queryBuilder
+                    ->andWhere($whereExpr[$i]);
+            }
+            $queryBuilder->setParameter($placeholder, $identifiersObjIterator->current());
 
-        foreach ($identifiers as $propertyName => $value) {
-            $placeholder = 'id_'.$propertyName;
-
-            $queryBuilder
-                ->where($queryBuilder->expr()->eq('o.'.$propertyName, ':'.$placeholder))
-                ->setParameter($placeholder, $value);
+            ++$i;
+            $identifiersObjIterator->next();
         }
```

Comment:
but to be fair, you could have another version where you check just if you're on the first property or not:

```php
private function processIdentifiers(array $identifiers, QueryBuilder $queryBuilder)
{
    if (0 === count($identifiers)) {
        return;
    }

    $firstIdentifier = reset($identifiers);
    foreach($identifiers as $identifier => $value) {
        $placeholder = ':id_'.$identifier;
        $expression = $queryBuilder->expr()->eq(
            'o.'.$identifier,
            ':'.$placeholder
        );

        if ($firstIdentifier === $identifier) {
            $queryBuilder->where($expression);
        } else {
            $queryBuilder->andWhere($expression);
        }
        
        $queryBuilder->setParameter($placeholder, $value);
    }
}
```